### PR TITLE
Update GCM lib and small fixes

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-10776

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
@dominicmarmionswrve @shanetmoore Could you have a quick look?

File: SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java [hunk 1]
```diff
@@ -128,8 +128,8 @@ public void setPushNotificationListener(ISwrvePushNotificationListener pushNotif
      * the Google Play Store or enable it in the device's system settings.
      */
     protected boolean checkPlayServices() {
-        int resultCode = GooglePlayServicesUtil.isGooglePlayServicesAvailable(context.get());
-        return resultCode == ConnectionResult.SUCCESS;
+        GoogleApiAvailability api = GoogleApiAvailability.getInstance();
+        return api.isGooglePlayServicesAvailable(context.get()) == ConnectionResult.SUCCESS;
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java]: Can we send to logcat a message containing the ConnectionResult value if its not success? Also rename it to isGooglePlayServicesAvailable?

File: SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java [hunk 2]
```diff
@@ -70,8 +70,8 @@ protected void beforeSendDeviceInfo(final Context context) {
             }
         }
 
-        // Google Advertising Id logging enabled
-        if (config.isGAIDLoggingEnabled()) {
+        // Google Advertising Id logging enabled and Google Play services ready
+        if (config.isGAIDLoggingEnabled() && checkPlayServices()) {
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java]: :+1:

File: SwrveSDK/build.gradle [hunk 3]
```diff
@@ -31,8 +31,8 @@ dependencies {
     compile 'com.google.code.gson:gson:2.2.+'
     compile 'com.android.support:support-v4:23.+'
     compile 'com.android.support:appcompat-v7:23.+'
-    googleCompile 'com.google.android.gms:play-services-gcm:8.1.0'
-    googleCompile 'com.google.android.gms:play-services-ads:8.1.0'
+    googleCompile 'com.google.android.gms:play-services-gcm:8.3.0'
+    googleCompile 'com.google.android.gms:play-services-ads:8.3.0'
```
Comment (shanethmoore) [inline SwrveSDK/build.gradle]: This is why I love gradle. So much nicer than Dling jars all over the place

File: SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java (hunk 1 above)
Comment (shanethmoore) [inline SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java]: If we do can we use SwrveLogger please?

Comment (Sergio-Mira) [inline SwrveSDK/src/google/java/com/swrve/sdk/Swrve.java]: Will do

**Comment by Sergio-Mira** [comment]
@dominicmarmionswrve @shanetmoore Updated to 8.4.0 and addressed feedback

**Comment by shanethmoore** [comment]
:rocket:

**Pull request merged**